src/ai/backend/agent/observer/heartbeat.py

```diff
@@ -33,34 +31,28 @@ def __init__(
     def name(cls) -> str:
         return "agent_hearbeat"
 
-    def _parse_container_data(
-        self, kernel_id: KernelId, container: Container
-    ) -> ContainerStatusData:
-        """
-        Helper method to parse container data into ContainerStatusData.
-        """
-        if kernel_obj := self._agent.kernel_registry.get(kernel_id):
-            kernel_status = str(kernel_obj.state)
-        else:
-            kernel_status = str(KernelLifecycleStatus.NOT_REGISTERED)
-        return ContainerStatusData(
-            container.id,
-            kernel_id,
-            container.status,
-            kernel_status,
-        )
-
     @override
     async def observe(self) -> None:
         containers = await self._agent.enumerate_containers(ContainerStatus.active_set())
 
         container_data = [
-            self._parse_container_data(kernel_id, container) for kernel_id, container in containers
+            ContainerStatusData(
+                container.id,
+                kernel_id,
+                container.status,
+            )
+            for kernel_id, container in containers
         ]
         await self._event_producer.anycast_event(
             AgentStatusHeartbeat(
                 self._agent.id,
                 container_data,
+                {
+                    kid: ContainerId(kernel_obj.container_id)
```

Comment:
Is the kernel ID used as the container ID value?

Reply:
It seems like we need to send the container ID value together with the kernel ID value.